step-templates/octopus-register-aks-k8s-cluster.json

```diff
@@ -0,0 +1,135 @@
 = $tenant.Replace(\" \", \"%20\")\n      $tenantUrl = \"$OctopusUrl/api/$SpaceId/tenants?skip=0&take=1000&name=$tenantEscaped\"\n      $tenantResponse = Invoke-RestMethod $tenantUrl -Headers $header \n\n      $tenantId = $tenantResponse.Items[0].Id\n      Write-Host \"The id for tenant $tenant is $tenantId\"\n      $tenantIdList += $tenantId\n    }\n}\n\n$machinePolicyId = $machinePolicyIdOrName\nif ($machinePolicyIdOrName.StartsWith(\"MachinePolicies-\") -eq $false)\n{\n\tWrite-Host \"The machine policy specified $machinePolicyIdOrName appears to be a name\"\n\t$machinePolicyNameEscaped = $machinePolicyIdOrName.Replace(\" \", \"%20\")\n\t$machinePolicyResponse = Invoke-RestMethod \"$OctopusUrl/api/$SpaceId/machinepolicies?partialName=$machinePolicyNameEscaped\" -Headers $header\n        \n    $machinePolicyId = $machinePolicyResponse.Items[0].Id\n    Write-Host \"The machine policy id is $machinePolicyId\"\n}\n\nif ([string]::IsNullOrWhiteSpace($machinePolicyId) -eq $true)\n{\n\tWrite-Host \"The machine policy $machinePolicyIdOrName cannot be found, exiting\"\n    exit 1\n}\n\n$workerPoolId = $WorkerPoolNameOrId\nif ([string]::IsNullOrWhiteSpace($workerPoolId) -eq $false -and $workerPoolId.StartsWith(\"WorkerPools-\") -eq $false)\n{\n\tWrite-Host \"The worker pool $workerPoolId appears to be a name, looking it up\"\n    $workerPoolNameEscaped = $workerPoolId.Replace(\" \", \"%20\")\n    $workerPoolResponse = Invoke-RestMethod \"$OctopusUrl/api/$SpaceId/workerpools?partialName=$workerPoolNameEscaped\" -Headers $header\n    \n    $workerPoolId = $workerPoolResponse.Items[0].Id\n    Write-Host \"The worker pool id is $workerPoolId\"\n}\n\n$rawRequest = @{\n\tId = $null;\n    MachinePolicyId = $MachinePolicyId;\n    Name = $RegistrationName;\n\tIsDisabled = $false;\n\tHealthStatus = \"Unknown\";\n\tHasLatestCalamari = $true;\n\tStatusSummary = $null;\n\tIsInProcess = $true;\n\tEndpoint = @{\n    \tId = $null;\n\t\tCommunicationStyle = \"Kubernetes\";\n\t\tLinks = $null;\n\t\tAccountType = \"AzureServicePrincipal\";\n        ClusterUrl = $null;\n        ClusterCertificate = $null;\n        SkipTlsVerification = $false;\n        DefaultWorkerPoolId = $workerPoolId;\n        Authentication = @{\n        \tAuthenticationType = \"KubernetesAzure\";\n            AccountId = $AzureAccountId;\n            ClusterName = $RegistrationName;\n            ClusterResourceGroup = $ClusterResourceGroup\n        };\n    };\n\tLinks = $null;\t\n\tRoles = $roleList;\n\tEnvironmentIds = $environmentIdList;\n\tTenantIds = $tenantIdList;\n    TenantedDeploymentParticipation = $DeploymentType;\n\tTenantTags = @()}\n\n$jsonRequest = $rawRequest | ConvertTo-Json -Depth 10\n\nWrite-Host \"Sending in the request $jsonRequest\"\n\n$machineUrl = \"$OctopusUrl/api/$SpaceId/machines\"\nWrite-Host \"Creating the target\"\n$machineResponse = Invoke-RestMethod $machineUrl -Headers $header -Method POST -Body $jsonRequest\n\nWrite-Host \"Create machine's response: $machineResponse\""
```

Comment:
Selfish but I'd like an overwrite option :)

Reply:
updated to include overwrite option for all three registration step templates